src/group_theory/perm/cycles.lean

```diff
@@ -500,22 +500,60 @@ begin
     exact support_pow_le _ n hx }
 end
 
-lemma is_cycle.pow_eq_one_iff [fintype α] {f : perm α} (hf : is_cycle f) {n : ℕ} :
-  f ^ n = 1 ↔ ∃ (x ∈ f.support), (f ^ n) x = x :=
+lemma is_cycle.pow_eq_one_iff [fintype β] {f : perm β} (hf : is_cycle f) {n : ℕ} :
+  f ^ n = 1 ↔ ∃ x, f x ≠ x ∧ (f ^ n) x = x :=
 begin
+  classical,
   split,
   { intro h,
     obtain ⟨x, hx, -⟩ := id hf,
-    exact ⟨x, mem_support.mpr hx, by simp [h]⟩ },
+    exact ⟨x, hx, by simp [h]⟩ },
   { rintro ⟨x, hx, hx'⟩,
     by_cases h : support (f ^ n) = support f,
-    { rw [←h, mem_support] at hx,
+    { rw [← mem_support, ← h, mem_support] at hx,
       contradiction },
     { rw [hf.support_pow_eq_iff, not_not] at h,
       obtain ⟨k, rfl⟩ := h,
       rw [pow_mul, pow_order_of_eq_one, one_pow] } }
 end
 
+lemma is_cycle.pow_eq_pow_iff [fintype β] {f : perm β} (hf : is_cycle f) {a b : ℕ} :
+  f ^ a = f ^ b ↔ ∃ x, f x ≠ x ∧ (f ^ a) x = (f ^ b) x :=
+begin
+  classical,
+  split,
+  { intro h,
+    obtain ⟨x, hx, -⟩ := id hf,
+    exact ⟨x, hx, by simp [h]⟩ },
+  { rintro ⟨x, hx, hx'⟩,
+    obtain hab | rfl | hab := lt_trichotomy a b,
```

Comment:
Can you use `wlog hab : a ≤ b` to deduplicate the proof?

Reply:
Sure, golfed the proof in the new commit